x/concentrated-liquidity/incentives_test.go

```diff
@@ -2747,3 +2759,231 @@ func (s *KeeperTestSuite) TestCollectIncentives() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestCreateIncentive() {
+	type testCreateIncentive struct {
+		poolId          uint64
+		isInvalidPoolId bool
+		sender          sdk.AccAddress
+		senderBalance   sdk.Coins
+		recordToSet     types.IncentiveRecord
+		existingRecords []types.IncentiveRecord
+
+		expectedError error
+	}
+	tests := map[string]testCreateIncentive{
+		"valid incentive record": {
+			poolId: defaultPoolId,
+			sender: s.TestAccs[0],
+			senderBalance: sdk.NewCoins(
+				sdk.NewCoin(
+					incentiveRecordOne.IncentiveDenom,
+					incentiveRecordOne.RemainingAmount.Ceil().RoundInt(),
+				),
+			),
+			recordToSet: incentiveRecordOne,
+		},
+		"record with different denom, emission rate, and min uptime": {
+			poolId: defaultPoolId,
+			sender: s.TestAccs[0],
+			senderBalance: sdk.NewCoins(
+				sdk.NewCoin(
+					incentiveRecordTwo.IncentiveDenom,
+					incentiveRecordTwo.RemainingAmount.Ceil().RoundInt(),
+				),
+			),
+			recordToSet: incentiveRecordTwo,
+		},
+		"record with different start time": {
+			poolId: defaultPoolId,
+			sender: s.TestAccs[0],
+			senderBalance: sdk.NewCoins(
+				sdk.NewCoin(
+					incentiveRecordOne.IncentiveDenom,
+					incentiveRecordOne.RemainingAmount.Ceil().RoundInt(),
+				),
+			),
+			recordToSet: withStartTime(incentiveRecordOne, defaultStartTime.Add(time.Hour)),
+		},
+		"record with different incentive amount": {
+			poolId: defaultPoolId,
+			sender: s.TestAccs[0],
+			senderBalance: sdk.NewCoins(
+				sdk.NewCoin(
+					incentiveRecordOne.IncentiveDenom,
+					sdk.NewInt(8),
+				),
+			),
+			recordToSet: withAmount(incentiveRecordOne, sdk.NewDec(8)),
+		},
+		"existing incentive records": {
+			poolId: defaultPoolId,
+			sender: s.TestAccs[0],
+			senderBalance: sdk.NewCoins(
+				sdk.NewCoin(
+					incentiveRecordOne.IncentiveDenom,
+					incentiveRecordOne.RemainingAmount.Ceil().RoundInt(),
+				),
+			),
+			recordToSet: incentiveRecordOne,
+			existingRecords: []types.IncentiveRecord{incentiveRecordTwo, incentiveRecordThree},
```

Comment:
Do we have a test case that tests what happens if someone creates the same incentive record twice? Does it add to the existing incentive record? I don't believe this tests that test case since I think incentiveRecordTwo and Three have different params.